In [1]:
import requests
import pandas as pd
import json

## Make an API request that returns the months where "fire" was searched in 2016. Which month had the most searches?

In [30]:
endpoint = 'https://data.nashville.gov/resource/fuaa-r5cm.json?year=2016&query_text=fire'

response = requests.get(endpoint) 

In [31]:
response

<Response [200]>

In [32]:
response.text

'[{"month_name":"January","year":"2016","query_count":"19","query_text":"fire"}\n,{"month_name":"February","year":"2016","query_count":"35","query_text":"fire"}\n,{"month_name":"March","year":"2016","query_count":"32","query_text":"fire"}\n,{"month_name":"April","year":"2016","query_count":"26","query_text":"fire"}\n,{"month_name":"May","year":"2016","query_count":"24","query_text":"fire"}\n,{"month_name":"June","year":"2016","query_count":"31","query_text":"fire"}\n,{"month_name":"July","year":"2016","query_count":"24","query_text":"fire"}\n,{"month_name":"August","year":"2016","query_count":"47","query_text":"fire"}\n,{"month_name":"September","year":"2016","query_count":"36","query_text":"fire"}\n,{"month_name":"October","year":"2016","query_count":"38","query_text":"fire"}\n,{"month_name":"November","year":"2016","query_count":"32","query_text":"fire"}\n,{"month_name":"December","year":"2016","query_count":"38","query_text":"fire"}]\n'

In [33]:
searches = response.json()
searches

[{'month_name': 'January',
  'year': '2016',
  'query_count': '19',
  'query_text': 'fire'},
 {'month_name': 'February',
  'year': '2016',
  'query_count': '35',
  'query_text': 'fire'},
 {'month_name': 'March',
  'year': '2016',
  'query_count': '32',
  'query_text': 'fire'},
 {'month_name': 'April',
  'year': '2016',
  'query_count': '26',
  'query_text': 'fire'},
 {'month_name': 'May',
  'year': '2016',
  'query_count': '24',
  'query_text': 'fire'},
 {'month_name': 'June',
  'year': '2016',
  'query_count': '31',
  'query_text': 'fire'},
 {'month_name': 'July',
  'year': '2016',
  'query_count': '24',
  'query_text': 'fire'},
 {'month_name': 'August',
  'year': '2016',
  'query_count': '47',
  'query_text': 'fire'},
 {'month_name': 'September',
  'year': '2016',
  'query_count': '36',
  'query_text': 'fire'},
 {'month_name': 'October',
  'year': '2016',
  'query_count': '38',
  'query_text': 'fire'},
 {'month_name': 'November',
  'year': '2016',
  'query_count': '32',
  'query_text

### At first it would only pull the 2014 data? i needed to get all of the years so I found where to filter data on the Socrata website.

ANSWER 1: August 2016 with 47 searches for fire

## 2. Make an API request that returns all the times a query was run more than 100 times in a month. How many times did this occur?

endpoint = 'https://data.nashville.gov/resource/fuaa-r5cm.json'

params = {
    'min': 100,
    'max': 600
}

response = requests.get(endpoint, params = params)

response.text

In [58]:
endpoint = 'https://data.nashville.gov/resource/fuaa-r5cm.json?$where=query_count>100&$limit=44000'

response = requests.get(endpoint) 

In [59]:
response.text

'[{"month_name":"March","year":"2014","query_count":"101","query_text":"permits"}\n,{"month_name":"January","year":"2015","query_count":"101","query_text":"criminal court clerk"}\n,{"month_name":"September","year":"2015","query_count":"101","query_text":"codes"}\n,{"month_name":"March","year":"2016","query_count":"101","query_text":"police"}\n,{"month_name":"March","year":"2016","query_count":"101","query_text":"civil service"}\n,{"month_name":"November","year":"2016","query_count":"101","query_text":"jobs"}\n,{"month_name":"November","year":"2017","query_count":"101","query_text":"metro holidays"}\n,{"month_name":"November","year":"2017","query_count":"101","query_text":"longevity pay"}\n,{"month_name":"January","year":"2018","query_count":"101","query_text":"West Nashville Heights Church of Christ"}\n,{"month_name":"January","year":"2018","query_count":"101","query_text":"property maps"}\n,{"month_name":"March","year":"2018","query_count":"101","query_text":"public works"}\n,{"month_

In [60]:
searches = response.json()
searches

[{'month_name': 'March',
  'year': '2014',
  'query_count': '101',
  'query_text': 'permits'},
 {'month_name': 'January',
  'year': '2015',
  'query_count': '101',
  'query_text': 'criminal court clerk'},
 {'month_name': 'September',
  'year': '2015',
  'query_count': '101',
  'query_text': 'codes'},
 {'month_name': 'March',
  'year': '2016',
  'query_count': '101',
  'query_text': 'police'},
 {'month_name': 'March',
  'year': '2016',
  'query_count': '101',
  'query_text': 'civil service'},
 {'month_name': 'November',
  'year': '2016',
  'query_count': '101',
  'query_text': 'jobs'},
 {'month_name': 'November',
  'year': '2017',
  'query_count': '101',
  'query_text': 'metro holidays'},
 {'month_name': 'November',
  'year': '2017',
  'query_count': '101',
  'query_text': 'longevity pay'},
 {'month_name': 'January',
  'year': '2018',
  'query_count': '101',
  'query_text': 'West Nashville Heights Church of Christ'},
 {'month_name': 'January',
  'year': '2018',
  'query_count': '101',
 

In [61]:
len(searches)

1262

ANSWER 2: 1262

## 3. Make another API request that returns all the times "codes" was searched more than 100 times in a month. How many times did this occur?

In [64]:
endpoint = 'https://data.nashville.gov/resource/fuaa-r5cm.json?$where=query_count>100&$limit=44000&query_text=codes'

response = requests.get(endpoint) 

In [65]:
response

<Response [200]>

In [66]:
response.text
searches = response.json()
searches

[{'month_name': 'September',
  'year': '2015',
  'query_count': '101',
  'query_text': 'codes'},
 {'month_name': 'April',
  'month': '4',
  'year': '2021',
  'query_count': '102',
  'query_text': 'codes'},
 {'month_name': 'August',
  'year': '2016',
  'query_count': '104',
  'query_text': 'codes'},
 {'month_name': 'November',
  'month': '11',
  'year': '2020',
  'query_count': '104',
  'query_text': 'codes'},
 {'month_name': 'December',
  'year': '2017',
  'query_count': '106',
  'query_text': 'codes'},
 {'month_name': 'May',
  'year': '2016',
  'query_count': '107',
  'query_text': 'codes'},
 {'month_name': 'April',
  'year': '2016',
  'query_count': '111',
  'query_text': 'codes'},
 {'month_name': 'July',
  'month': '7',
  'year': '2021',
  'query_count': '114',
  'query_text': 'codes'},
 {'month_name': 'January',
  'year': '2017',
  'query_count': '115',
  'query_text': 'codes'},
 {'month_name': 'May',
  'month': '5',
  'year': '2021',
  'query_count': '115',
  'query_text': 'codes'

In [68]:
len(searches)

56

ANSWER 3: 56